### PR TITLE
Fix sync method when called from threads without a context

### PR DESCRIPTION
**Pull request by hughsie**
Set the thread-default context for the current thread before performing the
async operation.

Hopefully fixes https://github.com/fwupd/fwupd/issues/2600

### CODE REVIEW

File: libfwupd/fwupd-client-sync.c [hunk 1]
```diff
@@ -1357,6 +1408,7 @@ fwupd_client_refresh_remote (FwupdClient *self,
 					   fwupd_client_refresh_remote_cb,
 					   helper);
 	g_main_loop_run (helper->loop);
+	g_main_context_pop_thread_default (helper->context);
```
Comment (superm1) [inline libfwupd/fwupd-client-sync.c:1411]: this is missing the matching `g_main_context_push_thread_default (helper->context);`

Comment (hughsie) [reply libfwupd/fwupd-client-sync.c:1413]: Eagle eyes! Fixed, thanks.

Comment (superm1) [reply libfwupd/fwupd-client-sync.c:1413]: I wish I could blame my eyes, but I tried it and saw it broken :)

**Review by superm1 — APPROVED** [review]
LGTM, and it fixes the underlying issue in #2600 it seems too (at least the reproducer doesn't fail anymore).

**Comment by aleixpol** [comment]
It does fix the issue, thank you very much!

Would you know under which version this will be fixed so I can tell our packagers to get that version? 1.5.2?

**Comment by hughsie** [comment]
1.5.2 indeed, which I can do next week if you remind me closer to the time. This week EOUTOFTIME :)

**Comment by aleixpol** [comment]
> 1.5.2 indeed, which I can do next week if you remind me closer to the time. This week EOUTOFTIME :)

No problem, will ping next week.

**Pull request merged**